aws/internal/keyvaluetags/key_value_tags.go

```diff
@@ -435,6 +472,30 @@ func (tags KeyValueTags) UrlEncode() string {
 	return values.Encode()
 }
 
+// MergeConfigTags creates KeyValueTags by merging KeyValueTags nested in
+// a configuration object with those represented as a map[string]interface{}.
+// Currently only supports the DefaultConfig type when passed to the interface{} argument.
+func MergeConfigTags(config interface{}, tags map[string]interface{}) KeyValueTags {
```

Comment:
objective here is to use this one-line method w/in resources so the the resource-level tags can be applied to those defined (or not) at the provider level without worrying about explicitly nil checking in the resource CRUD op
```go
defaultTagsConfig := meta.(*AWSClient).DefaultTagsConfig
tags := keyvaluetags.MergeConfigTags(defaultTagsConfig, d.Get("tags").(map[string]interface{}))
```

Reply:
i'm not too set on the method name though..

Reply:
My recommendation here is to instead create a method on `DefaultConfig`, which only works with the `KeyValueTags` type. 👍  This simplifies the implementation in the library and requires consumers to do any necessary type conversion before calling, which may change over time. e.g.

```go
func (dc *DefaultConfig) MergeTags(tags KeyValueTags) KeyValueTags {
	if dc == nil || dc.Tags == nil {
		return tags
	}

	return dc.Tags.Merge(tags)
}
```

With consumers doing:

```go
defaultTagsConfig := meta.(*AWSClient).DefaultTagsConfig
tags := defaultTagsConfig.MergeTags(keyvaluetags.New(d.Get("tags").(map[string]interface{})))
```

It looks like you have a similar method implemented above (just missing the `nil` check), so might just be a matter of updating implementation details.

Reply:
cool! i didnt know you could call a method on a nil object tbh 🤦‍♀️  ...i always assumed a nil pointer exception would be thrown in those cases